src/core/ext/transport/chttp2/transport/context_list.h

```diff
@@ -32,18 +36,59 @@ namespace grpc_core {
 class ContextList {
  public:
   // Creates a new element with \a context as the value and appends it to the
-  // list.
-  static void Append(ContextList** head, grpc_chttp2_stream* s);
+  // list. Each RPC/stream is associated with a unique \a context. This method
+  // is invoked when a chunk of data stored in an outgoing buffer is going to be
+  // sent over the wire. A data chunk being written over the wire is multiplexed
+  // with bytes from multiple RPCs. If one such RPC is traced, we store the
+  // following information about the traced RPC:
+  // - byte_offset_in_stream: Number of bytes belonging to that traced RPC which
+  // have been sent so far from the start of the RPC stream.
+  // - relative_start_pos_in_chunk: Starting offset of the traced RPC within
+  // the current chunk that is being sent.
+  // - num_traced_bytes_in_chunk: Number of bytes belonging to the traced RPC
+  // within the current chunk.
+  void Append(void* context, size_t byte_offset_in_stream,
+              int64_t relative_start_pos_in_chunk,
+              int64_t num_traced_bytes_in_chunk);
+
+  bool IsEmpty() { return context_list_entries_.empty(); }
 
-  // Executes a function \a fn with each context in the list and \a ts. It also
-  // frees up the entire list after this operation. It is intended as a callback
-  // and hence does not take a ref on \a error
+  // Interprets the passed arg as a pointer to ContextList type and executes the
+  // function set using grpc_http2_set_write_timestamps_callback method with
+  // each context in the list and \a ts. It also deletes/frees up the passed
+  // ContextList after this operation. It is intended as a callback and hence
+  // does not take a ref on \a error. The fn receives individual contexts in the
+  // same order in which they were Appended.
   static void Execute(void* arg, Timestamps* ts, grpc_error_handle error);
```

Comment:
This can also be made non-static like Append, correct? IIUC arg is always "this". I see this will require a cherry-pick to update google's init.cc together but this looks simple.

Reply:
Its no longer relevant now in the latest commit since we are directly dealing with the std::vector type.